### PR TITLE
Change `mbcs` encoding to `utf-8` in `com/win32com/client`

### PR DESCRIPTION
**Pull request by Avasam**
https://github.com/mhammond/pywin32/pull/1990#discussion_r1288601998
> [...] I suspect this is wrong now - when this file is being read I believe it will be assumed utf-8. Ditto the mbcs reference in build.py

\- @mhammond 

~~This changes more than what the initial comment mentioned, but I figured may as well review all usages of it. Even if we end up reverting because the needed changes would be too big or wrong.~~
- Changed `mbcs` to `utf-8` in `com/win32com/client`
- Removed unused `mbcs` encoding code in `Pythonwin/pywin/`

### CODE REVIEW

File: win32/Demos/win32gui_dialog.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def toparam(self):
                     # LVM_SETITEMW etc.
```
Comment (Avasam) [inline win32/Demos/win32gui_dialog.py:84]: > # Note this demo still works with byte strings.  An
> # alternate strategy would be to use unicode natively
> # and use the 'W' version of the messages - eg,

For instance this comment leads me to believe it won't be as simple as changing the encoding, unless it was specific to Python 2 ?

File: Pythonwin/pywin/tools/hierlist.py [hunk 2]
```diff
@@ -180,7 +180,7 @@ def AddItem(self, parentHandle, item, hInsertAfter=commctrl.TVI_LAST):
         if bitmapSel is None:
             bitmapSel = bitmapCol
         ## if isinstance(text, str):
-        ##  text = text.encode("mbcs")
+        ##  text = text.encode("utf-8")
```
Comment (mhammond) [inline Pythonwin/pywin/tools/hierlist.py:183]: These lines should just be removed.

File: isapi/install.py [hunk 3]
```diff
@@ -228,7 +228,7 @@ def FindWebServer(options, server_desc):
     # make sure server_desc is unicode (could be mbcs if passed in
     #  sys.argv).
     if server_desc and not isinstance(server_desc, str):
-        server_desc = server_desc.decode("mbcs")
+        server_desc = server_desc.decode("utf-8")
```
Comment (mhammond) [inline isapi/install.py:231]: as the comment notes, the source of this is sys.argv - is that now guaranteed to be utf-8? I suspect not.

**Review by mhammond — CHANGES_REQUESTED** [review]
I think for now we should only change the COM functions - where we generate a .py file. Most of the other places changed are quite possibly wrong as it's not Python file IO consuming the bytes.

File: Pythonwin/pywin/__init__.py [hunk 4]
```diff
@@ -4,7 +4,7 @@
 is_platform_unicode = 0
 
 # Ditto default_platform_encoding - not referenced and will die.
-default_platform_encoding = "mbcs"
+default_platform_encoding = "utf-8"
```
[the remark this answers is not in the document]
Comment (Avasam) [reply Pythonwin/pywin/__init__.py]: Indeed, not found anywhere.

**Review by mhammond — APPROVED** [review]
Thanks

**Comment by Avasam** [comment]
> Thanks

Bumping this PR. This has been approved but not merged @mhammond

**Pull request merged**